[FrameworkBundle] Remove env var table from AboutCommand

tuqqu opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | -
| Deprecations? | -
| Tickets       | -
| License       | MIT
| Doc PR        | -

Fixed AboutCommand output by shortening environment variable value (replacing with `...`) to fit the screen width if the value is too long.

Right now all output is a mess if it does not fit in the terminal window.

tuqqu committed:
[FrameworkBundle] Remove env var table from AboutCommand

ro0NL commented:
it's considerable to use the new Console dumper (#28898) for env  var values, that would improve display as well.

The CliDumper has `setMaxStringWidth`, so if we can pass such a value (as calculated here) we can depend on existing building blocks :)

something like

```diff
diff --git a/Command/AboutCommand.php b/Command/AboutCommand.php
index b9fbe67..feee4e6 100644
--- a/Command/AboutCommand.php
+++ b/Command/AboutCommand.php
@@ -12,6 +12,7 @@
 namespace Symfony\Bundle\FrameworkBundle\Command;
 
 use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Dumper;
 use Symfony\Component\Console\Helper\Helper;
 use Symfony\Component\Console\Helper\TableSeparator;
 use Symfony\Component\Console\Input\InputInterface;
@@ -90,12 +91,13 @@ EOT
         ];
 
         if ($dotenv = self::getDotenvVars()) {
+            $dumper = new Dumper($io);
             $rows = array_merge($rows, [
                 new TableSeparator(),
                 ['<info>Environment (.env)</>'],
                 new TableSeparator(),
-            ], array_map(function ($value, $name) {
-                return [$name, $value];
+            ], array_map(function ($value, $name) use ($dumper) {
+                return [$name, $dumper($value, $maxLength)];
             }, $dotenv, array_keys($dotenv)));
         }
```

tuqqu commented:
@ro0NL I'm not sure. `Dumper::__invoke()` does not allow to pass string's max length as the second parameter. Or are you suggesting to tweak its behaviour in this pr?

If so, `dump`'s output is standing out with it's bright colours and double quotes from all the rest. Do we really want this? 

If so, I could take a look into this.

ro0NL commented:
>  Or are you suggesting to tweak its behaviour in this pr?

yep :)

> If so, dump's output is standing out with it's bright colours and double quotes from all the rest. Do we really want this?

IMHO yes, but perhaps post a screenshot to see what others think. Maybe it's too much indeed ... i was mostly triggered for being able to reuse `CliDumper::setMaxStringWidth` :)

tuqqu commented:
@ro0NL I decided against using `Dumper` since we would rely on `CliDumper::setMaxStringWidth()` and it is part of `VarDumper` which is mostly a require-dev package, plus it adds a new dependency (on symfony/console). 

So I rewrote it a bit and it looks quite nice. I added a pic 
<img width="614" alt="Screenshot 2019-12-03 at 23 33 12" src="https://user-images.githubusercontent.com/47274443/70087417-568e7880-1625-11ea-857a-69b7e24c2bf1.png">

Can you make a review?

ro0NL commented on src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php:
```diff
@@ -141,4 +144,28 @@ private static function getDotenvVars(): array
+        return max(18/* longest column text */, ...array_map('strlen', array_keys($dotenvVars)));
+    }
+
+    private static function adjustLength(string $value, int $maxLength): string
```
`truncate()`

tuqqu replied:
renamed

ro0NL commented on src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php:
```diff
@@ -141,4 +144,28 @@ private static function getDotenvVars(): array
         return $vars;
     }
+
+    private static function getLongestColumnLength(array $dotenvVars): int
```
imho it's fine to inline this above

tuqqu replied:
fixed

ro0NL reviewed (approved):
> decided against using Dumper since we would rely on CliDumper::setMaxStringWidth() and it is part of VarDumper which is mostly a require-dev package

good point, i forgot we need to account for the fallback implementation as well.

LGTM in general :+1:

tuqqu commented:
@ro0NL fixed all

nicolas-grekas commented:
Not sure about this: while it improves the display, what if I *do* want to get the full value, eg to copy/paste it somewhere else?

ro0NL commented:
from experience, i agree `about` can be over-cluttered with output. Typically access tokens & co completely break the table

![image](https://user-images.githubusercontent.com/1047696/70861873-6d41a300-1f34-11ea-92d8-56b605bba678.png)

the original intend was always "debugging"; give a hint about the values being used. Truncating does not defeat this purpose IMHO.

> what if I do want to get the full value

`bin/console debug:container --env-var APP_SECRET` :)

nicolas-grekas commented:
Wouldn't it make more sense to just remove this table with env vars?
There is already `bin/console debug:container --env-vars` if one wants to display them.

ro0NL commented:
hmpf... yes. There's a subtle difference.. (envs managed by .env vs envs used in DI), but i agree it's not really significant.

Alternatively, what about displaying `SYMFONY_DOTENV_VARS` comma separeated in `about`, to not loose this feature (it's also done in the webprofiler e.g.)

ro0NL commented:
i tend to agree we should remove the Environment table in `about`. Actually the web profiler shows the server parameters used for that request (and highlights which are from .env)... this concept doesnt exists for the console.

The current table was a solution, prior to the later added `debug:container --env-vars`.

tuqqu commented:
@ro0NL @nicolas-grekas ok, I see. I pushed a new commit which removes env var table completely.

ro0NL reviewed (approved):
probably best to update the PR title

javiereguiluz reviewed (approved):
Arthur thanks for this contribution (and congrats for being your first Symfony contribution!). And thanks for your patience during the review process.

nicolas-grekas commented:
Thank you @tuqqu.
